### ISSUE TITLE
Fish Nightly started using old history

### ISSUE THREAD

**Issue by TimWolla** [fish-shell/fish-shell#3621]
**fish version installed** *(`fish --version`)*: fish, version 2.4.0-196-gbf53f39

**OS/terminal used**: Ubuntu, gnome-terminal

I'm running fish nightly and noticed that entries were missing in my scrollback history. Upon inspection I noticed that my `.local/share/fish/fish_history` had not been written to since yesterday noon (UTC+1). Instead the old `.config/fish/fish_history` file has been used.

Running fish using `strace -f -e open fish` and pressing the up key confirmed my suspicion:

```
open("/home/timwolla/.config/fish/fish_history", O_RDONLY|O_CLOEXEC) = 6
```

This is my XDG env:
```
[timwolla@~/.c/fish]env |grep XDG                                                                                                                                                                           21:30:47
XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/usr/share/upstart/xdg:/etc/xdg
XDG_CURRENT_DESKTOP=Unity
XDG_DATA_DIRS=/usr/share/ubuntu:/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/timwolla
XDG_RUNTIME_DIR=/run/user/1000
XDG_SEAT=seat0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_SESSION_DESKTOP=ubuntu
XDG_SESSION_ID=c2
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
XDG_SESSION_TYPE=x11
XDG_VTNR=7
```

I've since copied my history file from `.local/share` to `.config`.

**Comment by TimWolla** [comment]
Whoops. Ignore this one, apparently is missed an update. I noticed while looking through the commits myself. It is fixed (probably as of: 45d3a705bed3c7fc14ff8202356e745532a7682e)

**Comment by krader1961** [comment]
My apologies, @TimWolla, I broke it a week ago when refactoring the code to make the error messages more helpful when the config or data directory can't be used. Yes, commit 45d3a70 fixes that regression.
